### DISCUSSION TITLE
Does the bounded storage provider throttle the download?

### DISCUSSION THREAD

**Discussion by yara-blue** [aschey/stream-download-rs#26] [Q&A] [answered]
And if so how?

If I pause consuming my stream will the download continue or be paused too? 

A use case here would be a 3 hour long podcast (yeah they exist) that I pause after 30 minutes. The bounded storage would be set to support 30m of playtime. Would the bounded storage keep downloading in the background eventually making it impossible for me to resume?

**Comment by aschey** [comment, marked as answer]
Currently, there's no code that implements any throttling. I created the bounded storage module to handle something like internet radio where you could have a continuous stream that runs indefinitely, but it isn't designed to handle seeking, so you have no reason to keep a continuously growing stream of data when you only care about the last few MBs or so. For something like this, I think it would make sense to create a separate storage strategy.
